src/Altinn.Notifications.Core/INotificationsRepository.cs

```diff
@@ -10,7 +10,7 @@ public interface INotificationsRepository
 
         Task<Target> AddTarget(Target target);
 
-        Task<Target> UpdateSentTarget(int id);
+        Task<Target?> UpdateSentTarget(int id);
```

Comment:
Why introduce a nullable?

Reply:
I think it is the best way of transferring the knowledge that the target doesn't exist to the caller.

Reply:
But when we pass in an id, then it sounds more like an exception than that target does exists - I assume it must have existed at some point since we have an id - or am I missing something from the bigger picture?

Reply:
If this is used as signaling to the controller that the expected id passed in wasn't found then I would consider throwing an exception at this level and catch that in the controller with the reasoning that it is expected that the id is valid and should be there, it's really an exceptional case that it's not there.

Reply:
I'm not sure if Repository should assume anything of how it's methods are used or the context of the calls. I did consider letting the reader logic fail with an exception if there were no rows to read, but it didn't feel right in isolation.

Reply:
> I'm not sure if Repository should assume anything of how it's methods are used or the context of the calls.
Agree. That's basically why I'm proposing to throw an exception, because if an id is missing then something is wrong, unless records are wiped out on a regular basis. Then returning null makes more sense, or perhaps even a null object. Returning null forces a null check to be done on the caller side if you expect the record to be there in all/most situations. You said almost all cases, what are the cases when we don't expect to find an id?

> I did consider letting the reader logic fail with an exception if there were no rows to read, but it didn't feel right in isolation.
Why didn't it feel right?

Sorry for nagging on about this, but I quite don't get the case - and an id that earlier has been confirmed shouldn't just wannish.

Reply:
Current logic makes no assumption about whether the target exists. This is in line with the old code with the difference that the old code produced compile time warnings. The end result was the same.

I'm not entierly oposed to changing it. That was my initial idea. I went for the other way mostly because I was thinking that this is the best way to handle the missing target case.